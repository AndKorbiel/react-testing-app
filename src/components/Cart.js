import {Table, Button} from 'react-bootstrap';
export default function Cart(props) {
    return (
        <div>
            <h2>Your cart</h2>
            {props.list.length > 0 &&
            <Table striped bordered hover>
                <thead>
                <tr>
                    <th>#</th>
                    <th>Service name</th>
                    <th>Price</th>
                    <th></th>
                </tr>
                </thead>
                <tbody>
                    {props.list.map((el, index) => {
                        return(
                            <tr>
                                <td>{index + 1}</td>
                                <td>{el.name}</td>
                                <td>{el.price}</td>
                                <td><Button variant="danger" onClick={() => props.remove(el, index)}>X</Button></td>
                            </tr>
                        )
                    })}
                </tbody>
            </Table>
            }
        </div>
    )
}