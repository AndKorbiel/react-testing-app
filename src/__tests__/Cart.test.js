import React from 'react';
import Cart from "../components/Cart";
import {render, screen} from "@testing-library/react";
import renderer from 'react-test-renderer'

it('renders empty by default', ()=> {
    render(<Cart list={[]}/>)
    expect(screen.queryByRole('table')).not.toBeInTheDocument()
})

it('shows products on list with props', ()=> {
    render(<Cart list={[{name: 'Test service A'}, {name: 'Test product B'}]} />)
    expect(screen.queryByText(/Test service A/i)).toBeInTheDocument()
})

// snapshot tests
it('matches snapshot without props', ()=> {
    const tree = renderer
        .create(<Cart list={[]} />)
        .toJSON()
    expect(tree).toMatchSnapshot()
})

it('matches snapshot with props', ()=> {
    const tree = renderer
        .create(<Cart list={[{name: 'Test service A'}, {name: 'Test product B'}]} />)
        .toJSON()
    expect(tree).toMatchSnapshot()
})