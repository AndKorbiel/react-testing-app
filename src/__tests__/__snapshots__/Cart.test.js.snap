// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with props 1`] = `
<div>
  <h2>
    Your cart
  </h2>
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          Service name
        </th>
        <th>
          Price
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          Test service A
        </td>
        <td />
        <td>
          <button
            className="btn btn-danger"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            X
          </button>
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          Test product B
        </td>
        <td />
        <td>
          <button
            className="btn btn-danger"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            X
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`matches snapshot without props 1`] = `
<div>
  <h2>
    Your cart
  </h2>
</div>
`;
